pkgs/tools/system/incron/default.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchurl, bash }:
+
+stdenv.mkDerivation rec {
+  name = "incron-0.5.12";
+  src = fetchurl {
+    url = "https://github.com/ar-/incron/archive/0.5.12.tar.gz";
+    sha256 = "14cgsfyl43pd86wy40m1xwr7ww023n2jyks66ngybz5s4gbhps6c";
+  };
+
+  patchPhase = ''
+    sed -i "s|PREFIX = /usr/local|PREFIX = $out|g" Makefile
+    sed -i "s|/bin/bash|${bash}/bin/bash|g" usertable.cpp
+    sed -i "s|/usr/local/bin:/usr/bin:/bin:/usr/X11R6/bin|/run/current-system/sw/bin|g" usertable.cpp
```

Comment:
I have some concerns whether this is correct or not. For context see
https://github.com/ar-/incron/blob/master/usertable.cpp#L47 and
https://github.com/ar-/incron/blob/master/usertable.cpp#L605

Reply:
Yes, this seems weird. Also, here https://github.com/ar-/incron/blob/1eedfbc9b318372efd119fd17f4abdbde561a53d/usertable.cpp#L605

What's the expected behaviour here? Am I reading it right that it clears PATH and then sets it according to line 13? Is that sensible at all compared to actually inheriting PATH?

Reply:
@srhb my assumption is that since the user defined command runs as the user it doesn't seem entirely crazy to clear any unwanted  extras out of the environment. By adding PATH back in as the standard list of installed software the command can execute in a clean state I guess?

Is the path I used valid to grant the command access to the full list of software available to the user? Even if the path is valid does it make sense in the context of nixos? We want to balance changing the intended behavior of the software and what users would expect vs. what makes sense for nixos.

Reply:
Yes, this might be a case of conflicting logic especially with NixOS.

The path you added adds in the system profile, but not the user's profile, which is what seems like what incron should really be doing (or possibly both of them.) I'm not really sure how to do this in a sensible way aside from using `~/.nix-profile/bin` on PATH as well. I feel like it isn't correct to try to reconstruct the user's PATH by manually adding these things onto it. After all, that's what the normal login process would take care of, so I guess I don't have any terribly good ideas on how to deal with the assumptions in this program.

Reply:
@srhb I recall the issue now. The process is run as root and therefore it can't simply inherit PATH. So the issue becomes how to best reproduce what PATH would look like for a given user.

Reply:
@srhb I've modified the PATH to better match what NixOS provides as the user path... which as you mentioned is unfortunately best guess manually reconstructing the users PATH variable. I believe this new code to be most inline with what existing users might expect if using this software, and what is most aligned with the intention of the software developer.